test/tree.c

```diff
@@ -240,19 +241,19 @@ static void test_lookup_2(void)
 
 	c2 = nvme_lookup_ctrl(s, "tcp", "192.168.1.1", "192.168.1.20",
 			      "eth0", NULL, NULL);
-	assert(c1 == c2);
+	assert(c1 != c2);
```

Comment:
After a bit of pondering, I think this is just failing because `nvme_lookup_ctrl` is not able to lookup the IP configuration of the host via `getifaddrs`. And the new test case `test_ctrl_match_tcp` doesn't work on my system either. I suspect due to the same reason.

And another thing is that looking at `test_lookup_1` and `test_lookup_2` are doing not testing the right thing. Instead testing the host_traddr/host_iface variation they test the host_iface/trsvcid combination. As you explained in the excellent commit message we trsvcid is not optional for tcp.

Thus, I think we should 
- update/fix `test_lookup_1` and `test_lookup_2` so it tests the host_* combination for tcp and maybe have an individual test function for the different transports where these two arguments don't play a role.
- introduce a `getifaddrs` mock like we have for the `ioctl` so we do not depend on the host configuration at all.

What do you think?

Reply:
> * update/fix `test_lookup_1` and `test_lookup_2` so it tests the host_* combination for tcp and maybe have an individual test function for the different transports where these two arguments don't play a role.
> * introduce a `getifaddrs` mock like we have for the `ioctl` so we do not depend on the host configuration at all.
> 
> What do you think?

Good idea! I'll work on that.

Reply:
I removed `test_lookup_2()` and replaced by transport-specific tests: `test_ctrl_match_fc()`, `test_ctrl_match_rdma()`, `test_ctrl_match_tcp()`.